src/main/java/uk/gov/hmcts/reform/sendletter/services/encryption/PgpEncryptionUtil.java

```diff
@@ -44,43 +48,49 @@ private PgpEncryptionUtil() {
      * @param withIntegrityCheck Sets whether or not the resulting encrypted data will be protected
      *                           using an integrity packet.
      * @return PGP encrypted byte array
-     * @throws IOException  if an error occurs writing stream header information to the provider
-     *                      output stream.
-     * @throws PGPException if an error occurs initialising PGP encryption for the configured
-     *                      encryption method.
      */
     public static byte[] encryptFile(
         byte[] inputFile,
         String inputFileName,
         PGPPublicKey pgpPublicKey,
         boolean withIntegrityCheck
-    ) throws IOException, PGPException {
-        Security.addProvider(new BouncyCastleProvider());
-
-        ByteArrayOutputStream byteArrayOutputStream =
-            compressAndWriteFileToLiteralData(
-                inputFile,
-                inputFileName
-            );
-
-        PGPEncryptedDataGenerator encryptedDataGenerator =
-            prepareDataEncryptor(
-                pgpPublicKey,
-                withIntegrityCheck
-            );
-
-        return writeEncryptedDataToOutputStream(byteArrayOutputStream, encryptedDataGenerator);
+    ) {
+        try {
+            Security.addProvider(new BouncyCastleProvider());
+
+            ByteArrayOutputStream byteArrayOutputStream =
+                compressAndWriteFileToLiteralData(
+                    inputFile,
+                    inputFileName
+                );
+
+            PGPEncryptedDataGenerator encryptedDataGenerator =
+                prepareDataEncryptor(
+                    pgpPublicKey,
+                    withIntegrityCheck
+                );
+
+            return writeEncryptedDataToOutputStream(byteArrayOutputStream, encryptedDataGenerator);
+        } catch (IOException | PGPException exc) {
+            log.error("Error encrypting file", exc);
+            throw new UnableToPgpEncryptZipFileException(exc);
```

Comment:
I would go for more abstract name, e.g. `UnableToEncryptFileException`. There is no need to be extremely verbose with exception names